debugging/netcoreapp/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,249 @@
+
+        const containerName = debugConfiguration.dockerOptions && debugConfiguration.dockerOptions.run && debugConfiguration.dockerOptions.run.containerName
+            ? debugConfiguration.dockerOptions.run.containerName
+            : `${appName}-dev`; // TODO: Use unique ID instead?
+
+        const os = debugConfiguration.dockerOptions && debugConfiguration.dockerOptions.run && debugConfiguration.dockerOptions.run.os
+            ? debugConfiguration.dockerOptions.run.os
+            : 'Linux';
+
+        const appOutput = await this.inferAppOutput(debugConfiguration, os, resolvedAppProject);
+
+        const result = await this.dockerManager.prepareForLaunch({
+            appFolder: resolvedAppFolder,
+            appOutput,
+            build: {
+                context: resolvedContext,
+                dockerfile,
+                tag,
+                target
+            },
+            run: {
+                containerName,
+                os,
+            }
+        });
+
+        const configuration = this.createConfiguration(debugConfiguration, appFolder, result);
+
+        this.debugSessionManager.startListening();
+
+        return configuration;
+    }
+
+    private inferAppFolder(folder: WorkspaceFolder, configuration: DockerDebugConfiguration): string {
+        if (configuration.dockerOptions) {
+            if (configuration.dockerOptions.appFolder) {
+                return configuration.dockerOptions.appFolder;
+            }
+
+            if (configuration.dockerOptions.appProject) {
+                return path.dirname(configuration.dockerOptions.appProject);
+            }
+        }
+
+        return folder.uri.fsPath;
+    }
+
+    private async inferAppOutput(configuration: DockerDebugConfiguration, targetOS: PlatformType, resolvedAppProject: string): Promise<string> {
+        if (configuration.dockerOptions && configuration.dockerOptions.appOutput) {
+            return configuration.dockerOptions.appOutput;
+        }
+
+        const targetPath = await this.netCoreProjectProvider.getTargetPath(resolvedAppProject);
+        const relativeTargetPath = this.osProvider.pathNormalize(targetOS, path.relative(path.dirname(resolvedAppProject), targetPath));
+
+        return relativeTargetPath;
+    }
+
+    private async inferAppProject(configuration: DockerDebugConfiguration, resolvedAppFolder: string): Promise<string> {
+        if (configuration.dockerOptions) {
+            if (configuration.dockerOptions.appProject) {
+                return configuration.dockerOptions.appProject;
+            }
+        }
+
+        const files = await this.fsProvider.readDir(resolvedAppFolder);
+
+        const projectFile = files.find(file => path.extname(file) === '.csproj');
+
+        if (projectFile) {
+            return path.join(resolvedAppFolder, projectFile);
+        }
+
+        throw new Error('Unable to infer the application project file. It must be explicitly set in the Docker debug configuration.');
```

Comment:
You don't say how to do that or why it couldn't automatically be inferred.  My test project hits this.

Reply:
Added explicit instruction to set either the `appFolder` or `appProject` configuration property.